wrappers/swift/Logger.swift

```diff
@@ -0,0 +1,226 @@
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Wrapper class around ObjC Logger class `ODWLogger` used to events.
+public class Logger {
+    private var odwLogger: ODWLogger;
+
+    /// Constructs `Logger` with `ODWLogger` object which would be wrapped around.
+    public init(logger: ODWLogger) {
+        odwLogger = logger
+    }
+
+    // MARK: Basic LogEvent methods
+
+    /**
+    Logs a custom event with a specified name.
+
+    - Parameters:
+        - name: A `String` that contains the name of the event.
+    */
+    public func logEvent(name: String) {
+        self.odwLogger.logEvent(withName: name)
```

Comment:
nit: we don't need this `self.` nor the others in the functions that follow

Reply:
kept it intentionally as convention while referencing the instance vars.

Reply:
I understand keeping the convention. It's just not the Swift convention. In Swift, when we do `self.` it's a pattern signify value capturing through block :)